Documentation/Design/lambda_body_design.md

```diff
@@ -0,0 +1,71 @@
+
+## Language Examples
+
+### Java:  
+```Java
+(p, q) -> p == q
+```
+
+### Python:
+```Python
+lambda x, y: x == y 
+```
+
+### C#:
+```C#
+(x, y) => x == y;
+```
+
+### Ruby:
+```Ruby
+lambda { |x, y|  x == y }
+```
+
+### TypeScript
+```TypeScript
+(x, y) => x === y
+```
+
+
+## Design
+
+### GLS Syntax:
+`lambda : `*`[parameterName, parameterType, ...]`* `command`
+
+The GLS syntax for a lambda body will be as above. The command starts with `lambda :`, which is followed with zero or more parameter names, each one followed by its type. The final part is a GLS command. Any variables used in the GLS command must be passed in the parameter list that precedes the command.
+
+#### Examples:
+```
+lambda : x number y number { operation : x (equal to) y }
+```
+
+### Language Specific Properties:
+
+| Property Name               | Type    | Description                                                                           |
+|-----------------------------|---------|---------------------------------------------------------------------------------------|
+| lambdaLeft                  | string  | Language's syntax for the start of a lambda parmater list.                            |
+| lambdaMiddle                | string  | Language's syntax for the end of the lambda parameter list and the start of the body. |
+| lambdaRight                 | string  | Language's syntax for the end of the lambda body.                                     |
+| lambdaParameterTypeRequired | boolean | True if the language requires parameter types in the argument list, false otherwise.  |
+
+Command Format:
+```
+lambdaLeft parameterType, parameterName, ... lambdaMiddle commandString lambdaRight semicolon
+```
+
+The output starts with `lambdaLeft`. A list of parameters follows, comma separated. If the language property `lambdaParameterTypeRequired` is set to `false`, then all `parameterType`s are ommitted. `lambdaMiddle` follows the parameter list, followed by the actual code for the lambda. It is passed to this implmentation as a string which contains the output of another `command`. After the command string, a `lambdaRight` and `semicolon` end the output of the lambda command. 
+
+
+|              | lambdaLeft       | lambdaMiddle     | lambdaRight   | lambdaParameterTypeRequired   |
+|--------------|------------------|------------------|---------------|-------------------------------|
+| *Python*     |  `"lambda"`      |  `":"`           |  `""`         | `false`                       |
+| *C#*         |  `"("`           |  `") =>"`        |  `""`         | `false`                       |
+| *Java*       |  `"("`           |  `") ->"`        |  `""`         | `false`                       |
+| *Ruby*       |  `"lambda { \|"` |  `"\|"`          |  `"}"`        | `false`                       |
```

Comment:
The backslashes are required, otherwise markdown thinks they are part of the table. If you view it on github and see the markdown you should see a table with no backslashes. 


> On Apr 22, 2016, at 1:10 PM, Josh Goldberg <notifications@github.com> wrote:
> 
> In Documentation/Design/lambda_body_design.md:
> 
> > +| lambdaParameterTypeRequired | boolean | True if the language requires parameter types in the argument list, false otherwise.  |
> > +
> > +Command Format:
> > +```
> > +lambdaLeft parameterType, parameterName, ... lambdaMiddle commandString lambdaRight semicolon
> > +```
> > +
> > +The output starts with `lambdaLeft`. A list of parameters follows, comma separated. If the language property `lambdaParameterTypeRequired` is set to `false`, then all `parameterType`s are ommitted. `lambdaMiddle` follows the parameter list, followed by the actual code for the lambda. It is passed to this implmentation as a string which contains the output of another `command`. After the command string, a `lambdaRight` and `semicolon` end the output of the lambda command. 
> > +
> > +
> > +|              | lambdaLeft       | lambdaMiddle     | lambdaRight   | lambdaParameterTypeRequired   |
> > +|--------------|------------------|------------------|---------------|-------------------------------|
> > +| *Python*     |  `"lambda"`      |  `":"`           |  `""`         | `false`                       |
> > +| *C#*         |  `"("`           |  `") =>"`        |  `""`         | `false`                       |
> > +| *Java*       |  `"("`           |  `") ->"`        |  `""`         | `false`                       |
> > +| *Ruby*       |  `"lambda { \|"` |  `"\|"`          |  `"}"`        | `false`                       |
> Can you remove the backslashes? People will think you need to include \\ in the string.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>